### ISSUE TITLE
benchexec does not terminate when CTRL-c is pressed

### ISSUE THREAD

**Issue by dbeyer** [sosy-lab/benchexec#874]
In the phase at the very beginning, when benchexec computes the the set of benchmark tasks, it does not react on CTRL-c.

**Comment by PhilippWendler** [comment]
Is the problem that BenchExec does not react to Ctrl+C at all, or just that it reacts delayed, i.e., terminates after the initial phase is finished and actual benchmarking would start? I suspect it is the latter, because if we retrieve Ctrl+C we set an internal flag and that should be checked at latest when benchmarking starts.

So it would be a goal to reduce the delay somewhat, which would probably be possible by checking the mentioned flag at a few points during initialization.

**Comment by dbeyer** [comment]
Yes, exactly. For large benchmark sets, the initial phase can last several minutes, and sometimes one wants to terminate the process.
